numpy/lib/arraysetops.py

```diff
@@ -575,12 +575,28 @@ def in1d(ar1, ar2, assume_unique=False, invert=False):
     if len(ar2) < 10 * len(ar1) ** 0.145 or contains_object:
         if invert:
             mask = np.ones(len(ar1), dtype=bool)
-            for a in ar2:
-                mask &= (ar1 != a)
+            # If ar2.dtype is object, store is used to wrap the a value
```

Comment:
These new `if`/`else` statements should be redundant if you just append an additional axis to the object array in question. This way iteration will always yield your size-1 array.
``` diff
- ar2 = np.asarray(ar2).ravel()
+ ar2 = np.asarray(ar2).ravel()
+
+ # Ensure that iteration through object arrays yields size-1 arrays 
+ if ar2.dtype == object:
+     ar2 = ar2.reshape(-1, 1)
```

Reply:
Thanks for your input @BvB93. 
I'd love to remove the `if` / `else` but I'm not sure what you mean by appending an additional axis to the array. Could you explain a little more please?

Reply:
That was a great suggestion. Fixed the bug with a much simpler code.
Thanks for the help.

Reply:
Currently `ar2` is always a 1D array, so iterating through it will always yield its "scalar" elements 
(or tuples in case of your specific object array). By reshaping the shape-`(n,)` array into a shape-
`(n, 1)` array, iteration will now directly yield size-1 1D arrays without having to resort to this
temporary `store`.

``` python
In [28]: a = np.arange(3)

In [29]: a.shape
Out[29]: (3,)

In [30]: for i in a:  # yields scalars
    ...:     print(i)
    ...:
0
1
2

In [31]: b = a.reshape(-1, 1)

In [32]: b.shape
Out[32]: (3, 1)

In [33]: for i in b:  # yields size-1 1D arrays
    ...:     print(i)
    ...:
[0]
[1]
[2]
```